circuit_knitting/cutting/cutting_decomposition.py

```diff
@@ -260,3 +270,115 @@ def decompose_observables(
     }
 
     return subobservables_by_subsystem
+
+
+def find_cuts(
```

Comment:
I'd prefer to see this function introduced in a new/different file, since it's a bit independent of the existing code in this file.  (Actually, it depends on the existing code in this module through `cut_gates`, and moving to a separate file will help us make sure the dependency remains in a one-way direction, rather than have a risk of becoming spaghetti over time.)

Reply:
It was originally in `cut_finding` package, and I decided to move it up alongside the other modules in our cutting workflow. I chose `cutting_decomposition` because it felt like the most natural module. I've created a new module, `find_cuts.py` that includes this function now.